R/prep_scores_scorecard.R

```diff
@@ -47,3 +47,100 @@ prep_scores_scorecard <- function(results_portfolio,
+  } else {
+    tryCatch(
+      {
+        data_out <- emissions_data %>%
+          dplyr::filter(.data$entity == "portfolio") %>%
+          dplyr::mutate(
+            covered_aggregate_score = dplyr::if_else(
+              # sectors currently included in aggregate score calculation
+              .data$sector %in% c("Automotive", "Coal", "Oil&Gas", "Power", "Aviation", "Steel"),
+              TRUE,
+              FALSE
+            )
+          ) %>%
+          dplyr::group_by(covered_aggregate_score) %>%
+          dplyr::summarise(
+            emissions_pacta = sum(.data$weighted_sector_emissions, na.rm = TRUE),
+            .groups = "drop"
+          ) %>%
+          dplyr::ungroup() %>%
+          dplyr::mutate(
+            emissions_portfolio = sum(.data$emissions_pacta, na.rm = TRUE),
+            emissions_pacta_percent = emissions_pacta / emissions_portfolio
+          ) %>%
+          dplyr::filter(.data$covered_aggregate_score) %>%
+          dplyr::pull("emissions_pacta_percent")
+
+        stopifnot(length(data_out) == 1)
+
+        return(data_out)
+      },
+      error = function(e) {
+        pacta.portfolio.analysis:::write_log(
+          "ES: There was an error in prep_scores_emissions_scorecard().\nReturning NULL.\n",
+          file_path = log_dir
+        )
+        NULL
+      }
+    )
+  }
+}
+
+#' Prepare share of portfolio emissions covered by aggregate score analysis
+#'
+#' @param audit_data Data frame that contains pre-wrangled audit data
+#' @param currency_exchange_value Numeric vector with exchange rate
+#' @param total_portfolio_value_curr Numeric vector with total portfolio value
+#'   in target currency
+#'
+#' @return numeric
+#' @export
+prep_scores_exposure_scorecard <- function(audit_data,
+                                           currency_exchange_value,
+                                           total_portfolio_value_curr) {
+  if (any(is.null(emissions) | is.null(currency_exchange_value) | is.null(total_portfolio_value_curr))) {
+    data_out <- NULL
+    return(data_out)
+  } else {
+    tryCatch(
+      {
+        value_covered <- audit_data %>%
+          dplyr::filter(.data$entity == "portfolio") %>%
+          dplyr::filter(.data$asset_type %in% c("Bonds", "Equity")) %>%
+          # sectors currently included in aggregate score calculation
+          dplyr::filter(
+            .data$financial_sector %in% c("Automotive", "Coal", "Oil&Gas", "Power", "Aviation", "Steel")
+          ) %>%
+          dplyr::mutate(value_curr = .data$value_usd / .env$currency_exchange_value) %>%
+          dplyr::pull("value_curr") %>%
+          sum(na.rm = TRUE)
+
+        data_out <- value_covered / total_portfolio_value_curr
+
+        stopifnot(length(data_out) == 1)
+
+        return(data_out)
+      },
+      error = function(e) {
+        pacta.portfolio.analysis:::write_log(
+          "ES: There was an error in prep_scores_exposure_scorecard().\nReturning NULL.\n",
+          file_path = log_dir
```

Comment:
```suggestion
          file_path = .GlobalEnv$log_dir
```

Reply:
added new arg instead

Reply:
If you add a new arg, you need to make sure that that arg is passed now every time the function is called!
I think easiest might be to add the arg with a default value of `.GlobalEnv$log_dir`.